### PR TITLE
Mailing

### PR DESCRIPTION
**Pull request by UzunIon**

### CODE REVIEW

File: app/Http/Controllers/ContactUsController.php [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
 {
-    public function index() {
-        return view('contact-us');
+    public function index()
+    {
+        return view('contacts.contactUs');
+    }
+    public function send(ContactUsRequest $request): RedirectResponse
+    {
+        $callback = function (string $test): string{
```
Comment (egorbwork) [inline app/Http/Controllers/ContactUsController.php]: callback was removed in the end of Monday lesson. Please rewatch.

File: app/Http/Controllers/ContactUsController.php [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+            [
+                'email' => $data['email'],
+                'name' => $data['name'],
+                'phone' => $data['phone'],
+                'services' => $data['services'],
+                'messageText' => $data['message'],
+            ],
+            function(Message $message) use ($data){
+                $message->subject('Message from customer ' . $data['email']);
+                $message->to('tech@baloon.com');
```
Comment (egorbwork) [inline app/Http/Controllers/ContactUsController.php]: Please adjust to and from to use your project related values.

File: app/Http/Requests/ContactUsRequest.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+    public function rules(): array
+    {
+        return [
+            'email' => [ 'required', 'string', 'email'],
+            'name' => [ 'required', 'min:2', 'string' ],
```
Comment (egorbwork) [inline app/Http/Requests/ContactUsRequest.php]: I recommend to keep second type of data and next additional validations likes min

File: composer.json [hunk 4]
```diff
@@ -55,7 +56,13 @@
     "config": {
         "optimize-autoloader": true,
         "preferred-install": "dist",
-        "sort-packages": true
+        "sort-packages": true,
+        "platform": {
+            "php": "7.4.19"
```
Comment (egorbwork) [inline composer.json]: you should use php 8

File: resources/views/contacts/contactUs.blade.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            <form action="{{ route('contactUs.send') }}" method="POST" name="contact-form">
+                <div class="form-row">
+                    <div class="form-group col-md-6">
+                        <label for="email">Email</label>
+                        <input type="email" value="{{ old('email') }}" required="required" class="form-control" name="email" id="email" placeholder="example@example.com">
+                    </div>
+                    <div class="form-group col-md-6">
+                        <label for="name">Full Name</label>
+                        <input type="text" value="{{ old('name') }}" required="required" class="form-control" name="name" id="name" placeholder="John Doe">
+                        @error('name')
```
Comment (egorbwork) [inline resources/views/contacts/contactUs.blade.php:24]: I think it is extra, you already have block with all errors

**Review by egorbwork — CHANGES_REQUESTED** [review]
Looks good, please fix identified issues.

File: app/Http/Controllers/ContactUsController.php (hunk 1 above)
Comment (UzunIon) [reply app/Http/Controllers/ContactUsController.php]: Rewatched and removed callback function.

File: app/Http/Requests/ContactUsRequest.php (hunk 3 above)
Comment (UzunIon) [reply app/Http/Requests/ContactUsRequest.php]: Got it, I'll take that into account

File: app/Http/Controllers/ContactUsController.php (hunk 2 above)
Comment (UzunIon) [reply app/Http/Controllers/ContactUsController.php]: Adjusted for my project

File: resources/views/contacts/contactUs.blade.php (hunk 5 above)
Comment (UzunIon) [reply resources/views/contacts/contactUs.blade.php:24]: Is it better to have an error block for each separate input? Isn't it more user friendly?

File: composer.json (hunk 4 above)
Comment (UzunIon) [reply composer.json]: Ok

File: resources/views/contacts/contactUs.blade.php (hunk 5 above)
Comment (egorbwork) [reply resources/views/contacts/contactUs.blade.php:24]: if you removed the main block with errors, then yes. You can keep it.

**Review by egorbwork — APPROVED** [review]
Now looks workable. Nice job!

**Pull request merged**